client.go

```diff
@@ -232,15 +232,22 @@ func (c *Client) Get() error {
 		// Destination is the base name of the URL path in "any" mode when
 		// a file source is detected.
 		if mode == ClientModeFile {
-			filename := filepath.Base(u.Path)
+			filename, err := g.GetFilename(u)
```

Comment:
@schmichael , here use the new api Getter.Getfilename to allow Getters override the default filename.

by default `GetFilename` returns "" so that the other Getters fall back to original logic of computing filename

Reply:
Argh, sorry if I misled you, but I don't think adding a new method to the interface (and therefore every single implementation) is worth it if it's only implemented by the Maven getter.

Reply:
I think add this new method `GetFilename ` leaves flexibility for future in case new Getter need to control the default filename rather than passing query parameter "filename".

also, my original implementation of Getter.GetFile(dst, url) did not comply with the api as it does not save file to `dst`, instead, it's "dir(dst)/artifactId-version.type".
so two solution to fix this:
1) pass the correct filename in dst by passing the query parameter 'filename' in each url. (this is what i'm trying to avoid, especially when the default filename can be easily inferred by the MvnGetter)
2) do something in the interface to allow Getter control the default filename, the disadvantage is need to add single implementation to each Getter since golang not like java that support abstract class)

Reply:
@schmichael, what do you think about this?